clickhouse-admin/api/src/lib.rs

```diff
@@ -2,27 +2,46 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-use dropshot::{HttpError, HttpResponseOk, RequestContext};
+use clickhouse_admin_types::config::{KeeperConfig, ReplicaConfig};
+use clickhouse_admin_types::{KeeperSettings, ServerSettings};
+use dropshot::{
+    HttpError, HttpResponseCreated, Path, RequestContext, TypedBody,
+};
+use omicron_common::api::external::Generation;
 use schemars::JsonSchema;
-use serde::{Deserialize, Serialize};
-use std::net::SocketAddrV6;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct GenerationNum {
+    /// A unique identifier for the configuration generation.
+    pub generation: Generation,
+}
 
 #[dropshot::api_description]
 pub trait ClickhouseAdminApi {
     type Context;
 
-    /// Retrieve the address the ClickHouse server or keeper node is listening on
+    /// Generate a ClickHouse configuration file for a server node on a specified
+    /// directory.
     #[endpoint {
-        method = GET,
-        path = "/node/address",
+        method = POST,
+        path = "/node/server/generate-config/{generation}",
     }]
-    async fn clickhouse_address(
+    async fn generate_server_config(
         rqctx: RequestContext<Self::Context>,
-    ) -> Result<HttpResponseOk<ClickhouseAddress>, HttpError>;
-}
+        path: Path<GenerationNum>,
+        body: TypedBody<ServerSettings>,
+    ) -> Result<HttpResponseCreated<ReplicaConfig>, HttpError>;
 
-#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub struct ClickhouseAddress {
-    pub clickhouse_address: SocketAddrV6,
+    /// Generate a ClickHouse configuration file for a keeper node on a specified
+    /// directory.
+    #[endpoint {
+        method = POST,
+        path = "/node/keeper/generate-config/{generation}",
+    }]
+    async fn generate_keeper_config(
+        rqctx: RequestContext<Self::Context>,
+        path: Path<GenerationNum>,
+        body: TypedBody<KeeperSettings>,
+    ) -> Result<HttpResponseCreated<KeeperConfig>, HttpError>;
```

Comment:
Why are we returning the `KeeperConfig`?

Reply:
`KeeperConfig` and `ServerConfig` contain more settings than the ones that that can be configured via the API endpoint like ports etc. It'll be useful to have a response that contains all of the settings we're attempting to set for debugging purposes. In any case, it doesn't hurt to have a little more additional information? It's not a huge response. WDYT?

Reply:
Makes sense. Let's keep it.